### PR TITLE
Add support for has_one :x, polymorphic: true

### PR DESCRIPTION
**Pull request by ggordon**
This patch adds some support for polymorphic associations. This should address issue #77 when completed.

### CODE REVIEW

File: test/unit/serializer/polymorphism_test.rb [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+              self: '/pictures/1'
+            },
+            attributes: {
+              name: 'enterprise_gizmo.jpg'
+            },
+            relationships: {
+              imageable: {
+                links: {
+                  self: '/pictures/1/relationships/imageable',
+                  related: nil
```
Comment (lgebhardt) [inline test/unit/serializer/polymorphism_test.rb]: Is this the relationship related link you are questioning how to handle? Would `'/pictures/1/imageable'` work?

**Comment by ggordon** [comment]
Yes, that's the one. The problem with `/pictures/1/imageable` is that there is no `ImageableResource` or `ImageableController`, so sometimes, in this case, the imageable route would serialize a document and sometime a product, and I haven't figured out where that fits in. The problem with using `/pictures/1/document` is that it's _very difficult_ to know the all the possible routes to create.

**Comment by lgebhardt** [comment]
If we were to move the `get_related_resources` routes onto the parent resource's controller that would preclude the need for the polymorphic type controller and resource. Then `/pictures/1/imageable` could return resources of mixed type. We would need to expand support in the fields, sort, and filters to use the 'type dot field' notation and validate against the related resource types. That might require the polymorphic associations to know all the possible types, but I suspect we could find a work around to declaring that.

**Comment by ggordon** [comment]
I was surprised to find out that related resources work if empty `ImageableResource` and `ImageablesController` are created, definitely some magic beyond my understanding. Can even update a polymorphic association, but there is probably a better way to determine/set type on the association.

**Comment by lgebhardt** [comment]
That looks like it's working fairly well. A couple of questions:

What happens if you create a route to `imageable`? My guess is any actions directly on the ImageableController will fail as there isn't an imageable model. If that's the case I'm a bit worried it may be confusing to users.

Can we auto create the `imageable_type=` method on the resource when the association is polymorphic? Maybe with an override if the model attribute is different? I'm not that familiar with rails polymorphism, so maybe this isn't at all standard and thus a bad idea.

What else do you think needs to be added for completeness?

**Comment by ggordon** [comment]
Yeah, I'm not completely happy with having to create the `ImageablesController` and the routes would probably fail. I'll add a few more tests and clean up the code and docs and then we can reevaluate.

**Comment by barelyknown** [comment]
Hey @ggordon. I need to have this feature implemented very soon, so I'd be happy to help. How could I best assist you?

**Comment by ggordon** [comment]
@barelyknown Thanks for the offer. Is anything missing? As it stands now it supports the functionality I need. Let me know if I'm missing something.

**Comment by ggordon** [comment]
@lgebhardt My editor _cleaned up_ the Readme, and I didn't notice until after I pushed. Let me know if I should undo.

**Comment by lgebhardt** [comment]
@ggordon Yes, that would probably be good to undo the readme. What else changed in your latest commit?

**Comment by ggordon** [comment]
@lgebhardt Done. Added polymorphic_type method to Association, fixed the docs, and added some tests.

**Comment by barelyknown** [comment]
@lgebhardt How do you feel about this PR and https://github.com/cerebris/jsonapi-resources/pull/274?

**Comment by lgebhardt** [comment]
@barelyknown These look like a good start to supporting polymorphism. Sorry for the delay in merging them. I'll see if I can get them reviewed quickly once you address @dgeb's comments.

**Comment by ggordon** [comment]
@lgebhardt @barelyknown Should I close this PR? It looks like PR #274 includes this and more.

**Comment by barelyknown** [comment]
I think so.
